src/compute-client/src/controller/replica.rs

```diff
@@ -183,7 +188,10 @@ where
                     match ComputeGrpcClient::connect_partitioned(dests, version, client_params)
                         .await
                     {
-                        Ok(client) => Ok(SequentialHydration::new(client)),
+                        Ok(client) => {
+                            let dyncfg = self.dyncfg.clone();
```

Comment:
I don't have context on how often this clone and the `ReplicaClient::spawn` one get called, but if we're worried about the perf of cloning the entire internal BTreeMap, you could also wrap the ConfigSet in an Arc. Updating values and getting values both only require `&self`, the only thing you need `&mut self` for is adding new configs

Reply:
Good call! They should be called rarely enough, only when a replica is added or reconnects. But I like the idea of having everything in the compute controller share the same `ConfigSet` behind an `Arc`. Makes it cheap if we want to give it to more parts, and makes it obvious that the configs are shared.